### PR TITLE
migrator changes

### PR DESCRIPTION
**Pull request by NoahZinsmeister**
add ability to refund in eth

change slippage to v3 liquidity min

add refund tests

### CODE REVIEW

File: contracts/V3Migrator.sol [hunk 1]
```diff
@@ -13,19 +13,27 @@ import './libraries/TransferHelper.sol';
...
-        factory = _factory;
+    constructor(
+        address _factory,
+        address _WETH9,
+        address _nonfungiblePositionManager
+    ) PeripheryImmutableState(_factory, _WETH9) {
         nonfungiblePositionManager = _nonfungiblePositionManager;
     }
 
+    receive() external payable {
```
Comment (moodysalem) [inline contracts/V3Migrator.sol:33]: should this be inherited?

File: contracts/V3Migrator.sol [hunk 2]
```diff
@@ -43,7 +51,7 @@ contract V3Migrator is IV3Migrator, Multicall, SelfPermit {
 
     function migrate(MigrateParams calldata params) external override {
         // burn v2 liquidity to this address
-        IUniswapV2Pair(params.pair).transferFrom(msg.sender, params.pair, params.liquidityV2);
+        IUniswapV2Pair(params.pair).transferFrom(msg.sender, params.pair, params.liquidityToMigrate);
         (uint256 amount0V2, uint256 amount1V2) = IUniswapV2Pair(params.pair).burn(address(this));
 
         // get underlying tokens
```
Comment (moodysalem) [inline contracts/V3Migrator.sol:57]: probably cheaper to take v2Factory, token0, token1 and compute the pair, than make these two calls (although idk if the bytecode hash is same for all forks, probably not. should we prioritize uniswap gas cost tho?)

File: contracts/V3Migrator.sol [hunk 3]
```diff
@@ -85,21 +93,35 @@ contract V3Migrator is IV3Migrator, Multicall, SelfPermit {
...
                     recipient: params.recipient,
                     deadline: params.deadline
                 })
             );
 
-        // if necessary, refund dust and clear allowance
+        // if necessary, clear allowance and refund dust
         if (amount0V3 < amount0V2) {
-            TransferHelper.safeTransfer(token0, msg.sender, amount0V2 - amount0V3);
             IERC20(token0).approve(nonfungiblePositionManager, 0);
```
Comment (moodysalem) [inline contracts/V3Migrator.sol:105]: safeApprove?

**Review by moodysalem — APPROVED** [review]
sc code lgtm

File: contracts/V3Migrator.sol (hunk 1 above)
Comment (NoahZinsmeister) [reply contracts/V3Migrator.sol:33]: i don't want to add the sweep methods into here though, as we have an explicit param for handling eth and i don't want to confuse people

File: contracts/V3Migrator.sol (hunk 2 above)
Comment (NoahZinsmeister) [reply contracts/V3Migrator.sol:57]: for interop i prefer explicit pair address. perhaps we could just pass the token addresses in calldata?

Comment (moodysalem) [reply contracts/V3Migrator.sol:56]: yeah that works

**Pull request merged**